src/Command/InfectionCommand.php

```diff
@@ -205,17 +201,65 @@ protected function configure(): void
         ;
     }
 
-    protected function execute(InputInterface $input, OutputInterface $output)
+    protected function execute(InputInterface $input, OutputInterface $output): int
     {
         $this->startUp();
-        $this->runInitialTestSuite();
-        $this->runMutationTesting();
 
-        if (!$this->checkMetrics()) {
-            return 1;
-        }
+        /** @var CoverageRequirementChecker $coverageChecker */
+        $coverageChecker = $this->container['coverage.checker'];
+
+        /** @var Configuration $config */
+        $config = $this->container[Configuration::class];
+
+        /** @var Filesystem $fileSystem */
+        $fileSystem = $this->container['filesystem'];
+
+        /** @var TestFrameworkAdapter $adapter */
+        $adapter = $this->container[TestFrameworkAdapter::class];
+
+        /** @var SubscriberBuilder $subscriberBuilder */
+        $subscriberBuilder = $this->container['subscriber.builder'];
+
+        /** @var EventDispatcherInterface $eventDispatcher */
+        $eventDispatcher = $this->container['dispatcher'];
+
+        /** @var InitialTestsRunner $initialTestsRunner */
+        $initialTestsRunner = $this->container[InitialTestsRunner::class];
+
+        /** @var MemoryLimiter $memoryLimitApplier */
+        $memoryLimitApplier = $this->container['memory.limit.applier'];
+
+        /** @var MutationGenerator $mutationGenerator */
+        $mutationGenerator = $this->container[MutationGenerator::class];
 
-        return 0;
+        /** @var MutationTestingRunner $mutationTestingRunner */
+        $mutationTestingRunner = $this->container[MutationTestingRunner::class];
+
+        /** @var TestRunConstraintChecker $constraintChecker */
+        $constraintChecker = $this->container['test.run.constraint.checker'];
+
+        /** @var MetricsCalculator $metricsCalculator */
+        $metricsCalculator = $this->container['metrics'];
+
+        $engine = new Engine(
+            $coverageChecker,
+            $config,
+            $fileSystem,
+            $adapter,
+            $subscriberBuilder,
+            $eventDispatcher,
+            $initialTestsRunner,
+            $memoryLimitApplier,
+            $mutationGenerator,
+            $mutationTestingRunner,
+            $constraintChecker,
+            $this->consoleOutput,
+            $metricsCalculator
+        );
+
+        $result = $engine->execute((int) $this->input->getOption('threads'));
+
+        return (int) !$result;
```

Comment:
I'd prefer to make this slightly more explicit and use a ternary operator, e.g.:
```suggestion
        return $result ? 0 : 1;
```

Reply:
I'm not convinced tbh: I don't find using a ternary operator clearer plus it also involves a loose comparison which is not any better than a casting. Everything for a more verbose expression as well. But I'm making it to make you happy since it's not something I feel that strongly either